src/components/layouts/AdminLayout.tsx

```diff
@@ -0,0 +1,83 @@
+import {
+  Drawer,
+  List,
+  ListItemButton,
+  ListItemIcon,
+  ListItemText,
+} from "@mui/material";
+import InboxIcon from "@mui/icons-material/MoveToInbox";
+import MailIcon from "@mui/icons-material/Mail";
+import { useRouter } from "next/router";
+import { type Route } from "nextjs-routes";
+
+const DRAWER_WIDTH = 240;
+
+type MenuItem = {
+  text: string;
+  icon: JSX.Element;
+  path: "/" | Route;
+};
+
+export default function AdminLayout({
+  children,
+}: {
+  children: React.ReactNode;
+}) {
+  const router = useRouter();
+
+  const menuItems: MenuItem[] = [
```

Comment:
Can we move this out of the function and into the module scope? If not this object will be created each render unnecessarily.